### PR TITLE
Fix API to get language, Age at MRI and removed all the POST, PUT, PATCH requests from the API

### PR DESCRIPTION
**Pull request by cmadjar**
what was done for this PR:

1) Remove API v 0.0.1 and 0.0.2 to only support v0.0.3-dev as it is the version needed for the crawler

2) Removed all the POST, PUT, PATCH requests as well as any code that suggested modification of data in the database

3) Updated the documentation of the API that will be shared on the document repository of the open instance of PREVENT-AD for users to understand how to use the API
- removed documentation for POST, PUT, PATCH requests
- removed documentation for DDE since not pertinent in this instance
- removed EDC part of the documentation since not pertinent in this instance
- added Language to the candidate object
- added the age at MRI to the Visit object

### CODE REVIEW

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 1]
```diff
@@ -585,7 +511,6 @@ Returns file level QC information. It will return a JSON object of the form:
 }
 ```
 
-`PUT` requests to the same URL will update the QC information.
 
 ### 4.4 Alternate formats
```
Comment (cmadjar) [inline docs/API/LorisRESTAPI_v0.0.3.md]: TODO: modify the API so can download the nii files as well as a format for the imaging download

File: htdocs/api/v0.0.3-dev/Candidates.php [hunk 2]
```diff
@@ -115,78 +114,12 @@ public function handlePOST()
...
-            $projectName = $data['Candidate']['Project'];
-            $project     = \Project::singleton($projectName);
-            if (!empty($project)) {
-                \Candidate::singleton($candid)->setData(
-                    array('ProjectID' => $project->getId())
-                );
-            }
-        }
+        $this->header("HTTP/1.1 403 Forbidden");
+        $this->error("Create candidate: permission denied");
```
Comment (xlecours) [inline htdocs/api/v0.0.3-dev/Candidates.php]: This error message differ from the other.
Also, it<s not really a permission denied; it's more like 'Operation not permitted'

**Review by xlecours — APPROVED** [review]
I have tested the Language and Age at scan and it work fine. 
I have not tested all the POST/PUT/DELETE to see if they do not work.

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 3]
```diff
@@ -343,11 +298,9 @@ The JSON object is of the form:
 }
 ```
 
-where V1, V2, etc are the visit labels that are registered for this
-candidate.
+where V1, V2, etc are the visit labels that are registered for this candidate.
 
-PUT / PATCH are not supported for candidates in this version of the
-API.
+PUT / PATCH are not supported for candidates.
```
Comment (christinerogers) [inline docs/API/LorisRESTAPI_v0.0.3.md]: and POST?

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 4]
```diff
@@ -457,32 +400,16 @@ The format returned by a GET request is a JSON document of the form:
...
-to the URL SHOULD contain all fields with data entry. The server will null the data for keys not
-specified. A PUT request MAY not specify score columns that will be calculated/overwriten by
-server-side scoring. If the client attempted to score fields client-side and the value passed by the PUT
-request conflict with the server-side calculation of the score, the server-side calculation will win.
-Any keys specified in the document PUT that do not match a corresponding field in the form MAY be ignored.
-
-The specification for PATCH request is similar to a PUT request, with the exception that any
-fields not specified MUST be unmodified by the server rather than nulled. In most cases a series
-of PATCH requests SHOULD be used rather than a single PUT request for a client with pagination.
+PUT / PATCH / POST are not supported for this.
```
Comment (christinerogers) [inline docs/API/LorisRESTAPI_v0.0.3.md]: "this" -> "for instrument data"

Also, this should be the last line in the subsection, below the success/errors return (for GET)

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 5]
```diff
@@ -390,17 +344,12 @@ The JSON object is of the form:
 }
 ```
 
-A PUT of the same format but with only the Meta fields will create the VisitLabel
-for this candidate, in an unstarted stage if the Visit label provided is valid.
 
-PATCH is not supported for Visit Labels.
+POST / PUT / PATCH is not supported for Visit Labels.
 
 It will return a 404 Not Found if the visit label does not exist for this candidate
```
Comment (christinerogers) [inline docs/API/LorisRESTAPI_v0.0.3.md]: Switch lines 348 and 350: 
The 404 message applies to GET and so should stay with that text. 
Move the "POST not supported" line down below it.

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 6]
```diff
@@ -585,7 +511,6 @@ Returns file level QC information. It will return a JSON object of the form:
 }
```
Comment (christinerogers) [inline docs/API/LorisRESTAPI_v0.0.3.md]: Remove? PUT statement in 2 places above: 
- [ ] Line 537 (section 4.2)
- [ ] Line 497 (section 4.3.1)

**Review by christinerogers — CHANGES_REQUESTED** [review]
Hi Cecile, found small improvements and a few questions. 

The biggest question: can PUT still be used on the imaging QC ? (e.g. session level) 

In some places I could not comment directly on diff (because the line hadn't been changed), so I've indicated line number in the related comment (nearby but not exactly on the line).

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 7]
```diff
@@ -1,41 +1,33 @@
-# Loris API - v0.0.0-dev
+# Loris API - v0.0.3-dev
+
 ## 1.0 Overview
 
-This document specifies the Loris REST API.
+This document specifies the LORIS REST API.
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: Perfect! Done!

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 8]
```diff
@@ -1,41 +1,33 @@
...
 
 Any request sent to `$LorisRoot/api/$APIVERSION/$API_CALL` will return either a JSON object
-or no data. The Loris API uses standard HTTP error codes and the body of any response will
+or no data. The LORIS API uses standard HTTP error codes and the body of any response will
 either be empty or contain only a JSON object for any request.
 
 For brevity, the `$LorisRoot/api/$APIVERSION` is omitted from the definitions in this
 document. This document specifies $APIVERSION v0.0.3-dev and it
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: so far that is the case but we might want to remove it. I'll see

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 9]
```diff
@@ -1,41 +1,33 @@
...
 MUST be included before the request in all requests.
 
-HTTP GET requests NEVER modify data. PUT, POST or PATCH requests MUST be used to modify
-data as per their definitions in the HTTP/1.1 specification. Any methods not supported
+HTTP GET requests NEVER modify data. PUT, POST or PATCH requested are not supported
+in this instance of LORIS except for login. Any methods not supported
 will respond with a 405 Method Not Allowed response and an appropriate Allow header set (as
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: done

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 10]
```diff
@@ -1,41 +1,33 @@
...
 
 DELETE is not supported on any resource defined in this API.
 
 ### 1.1 Authentication
 
-If a user is logged in to Loris and can be authenticated using the standard session mechanism,
+If a user is logged in to LORIS and can be authenticated using the standard session mechanism,
 no further authentication is required. Requests will be evaluated as requests from that user,
-so that standard Loris modules can simply use the API.
+so that standard LORIS modules can simply use the API.
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: Good point! I decided to just remove the sentence. That is good to know for LORIS but a non-LORIS user trying to get the data for the open dataset will be WHAT?? ;). They don't need to know that the modules are using the API.

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 11]
```diff
@@ -1,41 +1,33 @@
...
 
-If a user is not logged in to Loris (for instance, in a third party app or a CORS application),
+If a user is not logged in to LORIS (for instance, in a third party app or a CORS application),
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: they might need that information when using the script or writing their own scripts if they want to do that.

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 12]
```diff
@@ -112,13 +99,13 @@ Where regex is a regular expression that can be used to validate a PSCID for thi
 If the type is "prompt", the user should be prompted to enter the PSCID for new candidates.
 If the type is "auto", the server will automatically generate the PSCID.
 
-Note that sometimes in Loris configurations "Site" is a part of the PSCID. This will be
+Note that sometimes in LORIS configurations "Site" is a part of the PSCID. This will be
 denoted by the string "SITE{1,1}" inside of the regex returned. This string should be replaced
 by the 3 letter site alias before attempting to pass this regex to a regular expression parser
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: they will see it in the front. Users won't know about the psc table or the config.xml since we are dealing with it. Since there is only one site, I will just hardcode it here for CONP and simplify the wording. Thanks!

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 13]
```diff
@@ -343,11 +298,9 @@ The JSON object is of the form:
 }
 ```
 
-where V1, V2, etc are the visit labels that are registered for this
-candidate.
+where V1, V2, etc are the visit labels that are registered for this candidate.
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: good point. They definitely are not, haha. Changing it

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 14]
```diff
@@ -457,32 +400,16 @@ The format returned by a GET request is a JSON document of the form:
...
 ```
 
-This URL is used to GET and modify flags for an instrument. The standard GET/PUT/PATCH
-rules apply. However, PATCH and PUT requests MUST include the "Meta" attribute and all
-fields in it MUST be specified and match the values in the URL, otherwise a "400 Bad request"
-error is returned and no data is modified. Like instruments, the [/dde] component is optional
-and used to differentiate single data entry and double data entry flags.
-
+This URL is used to GET flags for an instrument. The standard GET rules apply.
 The "Validity" flag may be missing, if the ValidityEnabled flag is not true for this instrument.
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: removed it since it will not be used in the open dataset

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 15]
```diff
@@ -800,13 +725,4 @@ object of the form:
 The `Modality` header in the SeriesInfo is either `MR` or `PT` for MRI or PET 
 scans, respectively.
 
-### 5.2 Tar Level Data
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: Haha, I was looking for it in the doc but I just realized that I removed that section so no need to change the header :P .

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 16]
```diff
@@ -64,36 +56,31 @@ for any future requests to authenciate the request.
 ## 2.0 Project API
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: done

File: docs/API/LorisRESTAPI_v0.0.3.md [hunk 17]
```diff
@@ -112,13 +99,13 @@ Where regex is a regular expression that can be used to validate a PSCID for thi
 If the type is "prompt", the user should be prompted to enter the PSCID for new candidates.
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: I removed that section since people won't be creating candidates and don't need that level of details I think. They would be more confused

File: docs/API/LorisRESTAPI_v0.0.3.md (hunk 5 above)
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: nice catch! Done

File: docs/API/LorisRESTAPI_v0.0.3.md (hunk 6 above)
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: done! Thanks!

File: docs/API/LorisRESTAPI_v0.0.3.md (hunk 3 above)
Comment (cmadjar) [reply docs/API/LorisRESTAPI_v0.0.3.md]: done

File: htdocs/api/v0.0.3-dev/Candidates.php (hunk 2 above)
Comment (cmadjar) [reply htdocs/api/v0.0.3-dev/Candidates.php]: good point! Will replace all instances with Operation not permitted
